application/core/LSYii_FilterValidator.php

```diff
@@ -36,6 +37,22 @@ protected function validateAttribute($object, $attribute)
         if ($this->skipOnEmpty && empty($object->$attribute)) {
             return;
         }
+
+        if (!empty($object->$attribute)) {
+            if ($object->$attribute == 'inherit') {
+                return;
+            }
+            elseif ($this->allowOnlyEmail) {
+                $emailRegex = '/^([a-zA-Z0-9.!#$%&’*+\/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+))(,([a-zA-Z0-9.!#$%&’*+\/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)))*$/';
+                $isValid = preg_match($emailRegex, (string) $object->$attribute);
+                if (!$isValid)
+                {
+                    Yii::app()->setFlashMessage(gT("Invalid email address."), 'error');
+                    Yii::app()->end();
+                }
+            }
+        }
+
```

Comment:
You don't need to create that functionality, it is already there in yii.
Just use 'filter' => 'email' in rules()
Then with the help of yii magic the CEmailValidator class is used for the validation.  Looks like it even uses the same regex expression.

Reply:
'filter' => 'email' is obviously not correct :)